geode-apis-compatible-with-redis/src/test/java/org/apache/geode/redis/internal/data/RedisSetTest.java

```diff
@@ -165,4 +174,142 @@ public void setExpirationTimestamp_stores_delta_that_is_stable() throws IOExcept
     o2.fromDelta(in);
     assertThat(o2).isEqualTo(o1);
   }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void should_calculateSize_closeToROS_withVaryingMemberCounts() {
+    for (int i = 0; i < 1024; i += 16) {
+      RedisSet set = createRedisSetOfSpecifiedSize(i);
+
+      int expected = reflectionObjectSizer.sizeof(set);
+      Long actual = Long.valueOf(set.getSizeInBytes());
+      Offset<Long> offset = Offset.offset(Math.round(expected * 0.06));
+
+      assertThat(actual).isCloseTo(expected, offset);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void should_calculateSize_closeToROS_withVaryingMemberSize() {
+    for (int i = 0; i < 16; i++) {
+      RedisSet set = createRedisSetWithMemberOfSpecifiedSize(i * 64);
+      int expected = reflectionObjectSizer.sizeof(set);
+      Long actual = Long.valueOf(set.getSizeInBytes());
+      Offset<Long> offset = Offset.offset(Math.round(expected * 0.06));
+
+      assertThat(actual).isCloseTo(expected, offset);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void size_shouldDecrease_WhenValueIsRemoved() {
+    Region region = mock(Region.class);
+    final RedisData returnData = mock(RedisData.class);
+    when(region.put(Object.class, Object.class)).thenReturn(returnData);
+    final RedisKey key = new RedisKey("key".getBytes());
+    final ByteArrayWrapper value = new ByteArrayWrapper("value".getBytes());
+
+    RedisSet set = createRedisSetOfSpecifiedSize(10);
+
+    ArrayList<ByteArrayWrapper> members = new ArrayList<>();
+    members.add(value);
+    set.sadd(members, region, key);
+
+    int initialSize = set.getSizeInBytes();
+
+    set.srem(members, region, key);
+
+    int finalSize = set.getSizeInBytes();
+    int expectedSize = initialSize - value.length() - PER_MEMBER_OVERHEAD;
+
+    assertThat(finalSize).isEqualTo(expectedSize);
+  }
+
+  /******** constants *******/
+  @Test
+  public void baseOverheadConstant_shouldMatchCalculatedValue() {
+    ReflectionObjectSizer reflectionObjectSizer = ReflectionObjectSizer.getInstance();
+    int baseRedisSetOverhead = reflectionObjectSizer.sizeof(new RedisSet()) + 18;
```

Comment:
I'm wondering how we came up with `18` here?  It is close to the size of an empty `ByteArrayWrapper` (which is 16)... not sure what the additional 2 would be... maybe padding?  Either way, a comment would be good explaining it.

Reply:
I also have no idea why it is 18. I didn't do the original Set implementation. figuring it out will take a lot of testing so I'd rather not have to. but I'll check with Ray on if he remembers where that came from.

Reply:
through a lot more work on RedisSet, I was able to get rid of the +18. the mysterious +5 is still there, but we now know this has something to do with the way the ByteArrayWrapper resizes. hopefully we can get rid of that magic number once we finish the story on getting rid of ByteArrayWrappers.